### PR TITLE
Add DEBUG_USD_SEED_DATA_PATH in config to make seed data flexible

### LINKED ISSUE #322: Make backend seed data more realistic
Currently, if `DEBUG_USE_SEED_DATA=True`, the backend fills the database with seed data on startup. This is meant for debugging. Now, as we get more into debugging higher-level features, we need more realistic seed data, such as diverse tasks, long messages, existing instances of most data types, etc.

### PR DESCRIPTION
**Pull request by kenhktsui**
Closes #322 

Factor out fixed seed data by adding DEBUG_USE_SEED_DATA_PATH in config to control seed data to use

### CODE REVIEW

**Review by andreaskoepf — APPROVED** [review]
LGTM! Thanks.

**Review by andreaskoepf — CHANGES_REQUESTED** [review]
The automatic contract-tests fail with during start of the mock-server with the following error-message:

```
 pydantic.error_wrappers.ValidationError: 1 validation error for Settings
DEBUG_USE_SEED_DATA_PATH
  file or directory at path "backend/test_data/generic/test_generic_data.json" does not exist (type=value_error.path.not_exists; path=backend/test_data/generic/test_generic_data.json)
 ```

Could you please check & fix?

File: backend/oasst_backend/config.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ class Settings(BaseSettings):
     DEBUG_ALLOW_ANY_API_KEY: bool = False
     DEBUG_SKIP_API_KEY_CHECK: bool = False
     DEBUG_USE_SEED_DATA: bool = False
+    DEBUG_USE_SEED_DATA_PATH: FilePath = "backend/test_data/generic/test_generic_data.json"
```
Comment (yk) [inline backend/oasst_backend/config.py:24]:
```suggestion
    DEBUG_USE_SEED_DATA_PATH: Optional[FilePath] = Path(__file__).parent.parent / "test_data/generic/test_generic_data.json"
```
needs `from pathlib import Path`

File: backend/main.py [hunk 2]
```diff
@@ -111,64 +112,10 @@ class DummyMessage(pydantic.BaseModel):
...
-                        user_message_id="0e276b98",
-                        parent_message_id="cec432cf",
-                        text="I'm unsure how to interpret this. Is it a riddle?",
-                        role="assistant",
-                    ),
-                ]
+                with open(settings.DEBUG_USE_SEED_DATA_PATH) as f:
+                    dummy_messages_raw = json.load(f)
+
+                dummy_messages = [DummyMessage(**dm) for dm in dummy_messages_raw]
```
Comment (yk) [inline backend/main.py:118]: maybe wrap this in a try block and do `logger.error` or `logger.exception` if it fails

**Review by yk — CHANGES_REQUESTED** [review]

File: backend/oasst_backend/config.py (hunk 1 above)
Comment (kenhktsui) [reply backend/oasst_backend/config.py:24]: Thanks. More safe to use pull path.

File: backend/main.py (hunk 2 above)
Comment (kenhktsui) [reply backend/main.py:118]: @yk I also thought so, but there is a broader try catch for seed data loading, so the program will still run without seed data with logging of exception and traceback if these lines got error.

**Comment by kenhktsui** [comment]
> The automatic contract-tests fail with during start of the mock-server with the following error-message:
> 
> ```
>  pydantic.error_wrappers.ValidationError: 1 validation error for Settings
> DEBUG_USE_SEED_DATA_PATH
>   file or directory at path "backend/test_data/generic/test_generic_data.json" does not exist (type=value_error.path.not_exists; path=backend/test_data/generic/test_generic_data.json)
> ```
> 
> Could you please check & fix?

Thanks for checking! The path of `test_data` now became absolute (dynamically) thanks to yk's suggestion. I could run `start-mock-server.sh` without error now.

**Review by andreaskoepf — APPROVED** [review]

**Review by yk — APPROVED** [review]

**Pull request merged**